sync/class.jetpack-sync-module-posts.php

```diff
@@ -316,76 +286,84 @@ function filter_post_content_and_add_links( $post_object ) {
 	}
 
 	public function save_published( $new_status, $old_status, $post ) {
-		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
-			$this->just_published[ $post->ID ] = true;
-		}
 
-		$this->previous_status[ $post->ID ] = $old_status;
+
+		if ( ! isset( $this->sync_items[ $post->ID ] ) ) {
+			$this->sync_items[ $post->ID ] = new Jetpack_Sync_Item( 'save_post' );
+		}
+		$sync_item = $this->sync_items[ $post->ID ];
+		$just_published = 'publish' === $new_status && 'publish' !== $old_status;
+		$sync_item->add_meta( 'just_published', $just_published );
+		$sync_item->add_meta( 'previous_status', $old_status );
 	}
 
 	public function wp_insert_post( $post_ID, $post = null, $update = null ) {
 		if ( ! is_numeric( $post_ID ) || is_null( $post ) ) {
 			return;
 		}
 
+		if ( wp_is_post_revision( $post ) && $this->is_saving_post( $post->post_parent ) ) {
+			$post = (array) $post;
+			unset( $post['post_content'] );
+			unset( $post['post_title'] );
+			unset( $post['post_excerpt'] );
+			$sync_item = $this->sync_items[ $post['post_parent']];
+			$sync_item->add_meta( 'revision', $post );
+			unset( $this->sync_items[ $post_ID ] );
+			return;
+		}
+
 		// workaround for https://github.com/woocommerce/woocommerce/issues/18007
 		if ( $post && 'shop_order' === $post->post_type ) {
 			$post = get_post( $post_ID );
 		}
-
-		$previous_status = isset( $this->previous_status[ $post_ID ] ) ?
-			$this->previous_status[ $post_ID ] :
-			self::DEFAULT_PREVIOUS_STATE;
-
-		$just_published = isset( $this->just_published[ $post_ID ] ) ?
-			$this->just_published[ $post_ID ] :
-			false;
-
-		$state = array(
-			'is_auto_save' => (bool) Jetpack_Constants::get_constant( 'DOING_AUTOSAVE' ),
-			'previous_status' => $previous_status,
-			'just_published' => $just_published
-		);
-		/**
-		 * Filter that is used to add to the post flags ( meta data ) when a post gets published
-		 *
-		 * @since 5.8.0
-		 *
-		 * @param int $post_ID the post ID
-		 * @param mixed $post WP_POST object
-		 * @param bool  $update Whether this is an existing post being updated or not.
-		 * @param mixed $state state
-		 *
-		 * @module sync
-		 */
-		do_action( 'jetpack_sync_save_post', $post_ID, $post, $update, $state );
-		unset( $this->previous_status[ $post_ID ] );
-		$this->send_published( $post_ID, $post );
-	}
-
-	public function send_published( $post_ID, $post ) {
-		if ( ! isset( $this->just_published[ $post_ID ] ) ) {
-			return;
+		if ( ! isset( $this->sync_items[ $post_ID ] ) ) {
```

Comment:
I think we can make the code even more accessible if we make an alias method that describes what we're actually testing for here. That is, instead of `if ( ! isset( $this->sync_items[ $post->ID ] ) ) {`, something like `if ( $this->is_new_post( $post->ID ) ) {`

Reply:
I agree